### PR TITLE
Update to the latest from observatorium/deployments

### PR DESCRIPTION
**Pull request by metalmatze**
There have been a couple of PRs against the deployments repository while I was moving the Operator to this separate repository. Here is what is still missing from those.

https://github.com/observatorium/deployments/pull/331
https://github.com/observatorium/deployments/pull/332
https://github.com/observatorium/deployments/pull/333
https://github.com/observatorium/deployments/pull/334

### CODE REVIEW

File: example/main.jsonnet [hunk 1]
```diff
@@ -1,4 +1,4 @@
-local dex = (import 'github.com/observatorium/deployments/components/dex.libsonnet') + {
+local dex = (import '../components/dex.libsonnet') + {
```
Comment (squat) [inline example/main.jsonnet:1]: we don't expect there to be any `../components` here, right? it should be `deployments/components` or something, no?

Comment (metalmatze) [reply example/main.jsonnet:1]: I really want to start pushing the absolute paths everywhere by the way. :wink: 
So going forward, if you see any, please help clean them up. :+1:

File: main.jsonnet [hunk 2]
```diff
@@ -0,0 +1,172 @@
+local dex = (import '../components/dex.libsonnet') + {
```
Comment (squat) [inline main.jsonnet:1]: how is this working with `../`?

File: main.jsonnet [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    },
+    version: 'v2.24.0',
+    image: 'quay.io/dexidp/dex:v2.24.0',
+    commonLabels+:: {
+      'app.kubernetes.io/instance': 'e2e-test',
+    },
+  },
+};
+
+local obs = (import '../environments/base/observatorium.jsonnet');
```
Comment (squat) [inline main.jsonnet:36]: and this?

File: main.jsonnet (hunk 2 above)
Comment (squat) [reply main.jsonnet:1]: Is this whole main.jsonnet simply unused maybe?

File: example/main.jsonnet [hunk 4]
```diff
@@ -33,7 +33,7 @@ local dex = (import 'github.com/observatorium/deployments/components/dex.libsonn
   },
 };
 
-local obs = (import 'github.com/observatorium/deployments/environments/base/observatorium.jsonnet');
+local obs = (import '../environments/base/observatorium.jsonnet');
```
Comment (squat) [inline example/main.jsonnet:36]: how is this working?

**Comment by squat** [comment]
I'm glad the tests are passing :)) but I'm a bit confused about some things

File: jsonnet/obs-operator.jsonnet [hunk 5]
```diff
@@ -1,9 +1,9 @@
 local t = (import 'github.com/thanos-io/kube-thanos/jsonnet/kube-thanos/thanos.libsonnet');
+local loki = import 'github.com/observatorium/deployments/components/loki.libsonnet';
 local config = import './operator-config.libsonnet';
-local obs = ((import 'vendor/github.com/observatorium/deployments/components/observatorium.libsonnet') + {
+local obs = ((import 'github.com/observatorium/deployments/components/observatorium.libsonnet') + {
```
Comment (clyang82) [inline jsonnet/obs-operator.jsonnet:4]: I am not sure if this way can be working in DS build. Maybe we need to have it vendorize as well.

**Comment by clyang82** [comment]
I just want to ask maybe a stupid question here: What is purpose of `observatorium/deployments`? is the purpose?
- observatorium/deployments is designed to be installed by simple `kubectl apply -f ...`
- observatorium/operator is designed to be installed by operator

File: jsonnet/obs-operator.jsonnet (hunk 5 above)
Comment (metalmatze) [reply jsonnet/obs-operator.jsonnet:4]: This is vendored. It is working just like before. The `jsonnet -J vendor/` flag tells jsonnet to import the files from the vendor folder and in there is a `github.com/observatorium/...` folder. Nothing really changed, except not using the symlinks anymore :)

**Comment by metalmatze** [comment]
> * observatorium/deployments is designed to be installed by simple `kubectl apply -f ...`
> * observatorium/operator is designed to be installed by operator

That's exactly right. We still depend on only the deployments repository to deploy the stack without the operator ourselves, as we cannot simply install new CustomResourceDefinitions on our cluster.

File: main.jsonnet (hunk 2 above)
Comment (metalmatze) [reply main.jsonnet:1]: Yes, this needs to be fixed and we need to add a test case that checks if this compiles correctly.

File: main.jsonnet (hunk 3 above)
Comment (metalmatze) [reply main.jsonnet:36]: Yes, this needs to be fixed and we need to add a test case that checks if this compiles correctly.

File: example/main.jsonnet (hunk 4 above)
Comment (metalmatze) [reply example/main.jsonnet:36]: Yes, this needs to be fixed and we need to add a test case that checks if this compiles correctly.

**Review by squat — APPROVED** [review]
Nice!

**Pull request merged**